src/components/DeploymentsList.vue

```diff
@@ -15,15 +15,7 @@
       </template>
     </p-list-header>
 
-    <p-table :data="deployments" :columns="columns" class="deployments-list__table">
-      <template #selection-heading>
-        <p-checkbox v-model="model" @update:model-value="selectAllDeployments" />
-      </template>
-
-      <template #selection="{ row }">
-        <p-checkbox v-model="selectedDeployments" :value="row.id" />
-      </template>
-
+    <p-table :selected="can.delete.deployment ? selectedDeployments : undefined" :data="deployments" :columns="columns" class="deployments-list__table" @update:selected="selectedDeployments = $event">
```

Comment:
Don't _love_ having to break up the v-model here. 

With a computed `selectedDeploymentsModel` with a setter/getter though, the value becomes possibly undefined which means a bunch of `?.` need to get added around whereas this isolates the nullish-ness. 

It also ends up being more boilerplatey code and see further down in `WorkPoolQueuesTable` where something like that was done before and it led to an innocuous little bug 🤷 .